### PR TITLE
[SYCL] Add <iostream> headers to tests which assumed "CL/sycl.hpp" would have it included

### PR DESCRIPTION
**Pull request by raaiq1**
The purpose of this PR is support the changes made in PR https://github.com/intel/llvm/pull/6469 , which gets rid of `#include <iostream>` from sycl headers and is assumed to be included in the "sycl/sycl.hpp" header for some of the tests present in this repository

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 /test-suite-externals
 *.pyc
 /build*
+.vscode/**
```
Comment (cperkinsintel) [inline .gitignore]: the .gitignore file probably shouldn't be modified for everyone in this fashion.

I'm not sure why you'd have a .vscode file inside your repo directory. If using VSCode to "open" a directory, make sure it's the parent.

**Review by cperkinsintel — DISMISSED** [review]
one nit, otherwise LGTM

**Review by cperkinsintel — DISMISSED** [review]
LGTM

**Review by aelovikov-intel — DISMISSED** [review]
IMO, this should be committed ASAP before waiting for the intel/llvm PR.

The failure looks like it might have been cause by this patch though.

File: SYCL/Regression/device_num.cpp [hunk 2]
```diff
@@ -68,8 +69,9 @@ void PrintSystemConfiguration() {
     auto PlatformName = Platform.get_info<info::platform::name>();
     const auto &Devices = Platform.get_devices();
     for (const auto &Device : Devices) {
-      std::cout << Backend << ":" << getDeviceTypeName(Device) << ":"
-                << DeviceNums[Backend] << std::endl;
+      std::cout << backend_to_string(Backend) << ":"
```
Comment (aelovikov-intel) [inline SYCL/Regression/device_num.cpp]: I would split this into a separate PR so that the current one only had `#include <iostream>` and could be committed without the main patch.

File: SYCL/ESIMD/api/simd_any_all.cpp [hunk 3]
```diff
@@ -111,7 +112,8 @@ template <typename T, int N, int Op> bool test_impl(queue q) {
     }
     T gold = Gold[Op * num_vals + i];
     T val = res[i];
-    std::cout << "  " << ops[Op] << "(" << (simd<ValTy, N>)test_vals_arr[i]
+    std::cout << "  " << ops[Op] << "("
```
Comment (aelovikov-intel) [inline SYCL/ESIMD/api/simd_any_all.cpp]: Split into a separate PR - same as my another comment.

**Comment by aelovikov-intel** [comment]
> The purpose of this PR is support the changes made in draft PR [intel/llvm#6337](https://github.com/intel/llvm/pull/6337) , which gets rid of `#include <iostream>` from sycl headers and is assumed to be included in the "CL/sycl.hpp" header for some of the tests present in this repository

Need to update PR and the header (`sycl/sycl.hpp`).

**Review by aelovikov-intel — DISMISSED** [review]

**Comment by aelovikov-intel** [comment]
The changes are good. Please update the first comment so that resulting commit message is accurate.

**Comment by raaiq1** [comment]
@intel/llvm-gatekeepers slight ping to merge commit. Thanks

**Review by againull — DISMISSED** [review]

**Review by asudarsa — DISMISSED** [review]
LGTM. No red flags..:-). Thanks

**Review by v-klochkov — DISMISSED** [review]

**Comment by pvchupin** [comment]
I had to make some trivial conflict resolutions due to cl::sycl -> sycl namespace change done earlier
